symbolic-base/src/ZkFold/Symbolic/Data/Eq.hs

```diff
@@ -62,3 +69,24 @@ instance (Symbolic c, Haskell.Eq (BaseField c), Representable f, Traversable f)
         in
             any Bool (unpacked result)
 
+instance Symbolic c => Eq (Bool c) (Bool c)
```

Comment:
`Eq (Bool c) (Bool c)` can be implemented in a single constraint instead of two in a Generic implementation

Reply:
Ok, let's have it in a separate PR